packages/esm-active-visits-app/src/visits-summary/visits-components/encounter-list.component.tsx

```diff
@@ -0,0 +1,114 @@
+import React, { useEffect, useState } from 'react';
+import DataTable, {
+  TableContainer,
+  Table,
+  TableHead,
+  TableExpandHeader,
+  TableRow,
+  TableHeader,
+  TableBody,
+  TableExpandRow,
+  TableCell,
+  TableExpandedRow,
+} from 'carbon-components-react/es/components/DataTable';
+import EncounterObservations from './encounter-observations.component';
+import styles from '../visit-detail-overview.scss';
+import { Observation } from '../visit.resource';
+import { useTranslation } from 'react-i18next';
+import { useLayoutType } from '@openmrs/esm-framework';
+
+interface EncounterListProps {
+  encounters: Array<{
+    id: any;
+    time: any;
+    encounterType: string;
+    provider: string;
+    obs: Array<Observation>;
+  }>;
+  visitUuid: string;
+}
+
+const EncounterListDataTable: React.FC<EncounterListProps> = ({ encounters, visitUuid }) => {
+  const { t } = useTranslation();
+  const layout = useLayoutType();
+  const isDesktop = layout === 'desktop';
+  const [headerWidth, setHeaderWidth] = useState(0);
+
+  const headerData = [
+    {
+      id: 1,
+      header: 'Time',
```

Comment:
Localize these headers.
Memoize the headerData.

Reply:
I didn't understand `Localize these headers`. Can you please explain more briefly?
Thanks!

Reply:
Right now the headers are all English. They need to be `t(...)`'ed to appear in other languages.

Reply:
Example:

```js
const headerData = useMemo(() => [
  { id: 1, header: t('timeHeader', 'Time'), key: 'time' },
  // ...
], [t]);
```

Reply:
Ohh okay! Got it now. Thanks alot!